src/main/java/com/ProjectEureka/backend/impls/UserServiceImpls.java

```diff
@@ -0,0 +1,75 @@
+package com.ProjectEureka.backend.impls;
+
+import java.util.List;
+import java.util.Optional;
+
+import com.ProjectEureka.backend.models.User;
+import com.ProjectEureka.backend.repositories.UserRepository;
+import com.ProjectEureka.backend.services.UserService;
+import org.springframework.stereotype.Service;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.data.rest.webmvc.ResourceNotFoundException;
+
+@Service
+public class UserServiceImpls implements UserService {
+    @Autowired
+    UserRepository userRepository;
+
+    @Override
+    public User createUser(User user) {
+        return userRepository.save(user);
+    }
+
+    @Override
+    public List<User> getAllUsers() {
+        return userRepository.findAll();
+    }
+
+    @Override
+    public User getUserById(String id) {
+        Optional<User> user = this.userRepository.findById(id);
+
+        if (user.isPresent()) {
+            return user.get();
+        } else {
+            throw new ResourceNotFoundException("User not found with id: " + id);
+        }
+    }
+
+    @Override
+    public User deletedUser(String id) {
+        Optional<User> user = this.userRepository.findById(id);
+
+        if (user.isPresent()) {
+            User deletedUser = user.get();
+            this.userRepository.delete(user.get());
+            return deletedUser;
```

Comment:
these `deletedUser` are fine. Just rename the method names that have `deletedUser` to just `deleteUser`